backend/src/routes/api/getUserProfileById.js

```diff
@@ -0,0 +1,22 @@
+import { StatusCodes } from 'http-status-codes';
+import Profile from '../../models/Profile.js';
+
+const userProfileEndpoint = async (router) => {
+  router.get('/users/:id', (req, res) => {
+    let id = req.params.id;
```

Comment:
this variable shouldn't change, so let's change it to const id

additionally, just "id" may be confusing in bigger requests (userId would be better name for this)